Chunk store test integrated

krishnaIndia opened the pull request:
Have implemented the initial utility functions and structured it. Will continue writing the helper functions and then the unit tests.

dirvine commented on src/chunk_store/test.rs:
```diff
@@ -0,0 +1,82 @@
+use std::path::Path;
+use std::fs::{remove_dir_all};
+
+static ONE_KB: u64 = 1024u64;
```
I think this can be just `static ONE_KB: u64 = 1024;`

dirvine commented on src/chunk_store/test.rs:
```diff
@@ -0,0 +1,82 @@
+use std::fs::{remove_dir_all};
+
+static ONE_KB: u64 = 1024u64;
+static AES_PADDING: u64 = 16u64;
```
We do no tneed to consider encrypting these chunks in the chunk-store any more. They will be session based so we can store them as they are (already encrypted anyway). Keeps it simpler. So this becomes literally a HashMap of <NameType, Data> . Should simplify it a lot :-)

krishnaIndia commented:
Updated the unit tests. Two more unit tests are pending. Unused code to be removed after the tests are wired up. Implemented `get` in chunk_store.rs for writing the tests.

dirvine commented on Cargo.toml:
```diff
@@ -25,3 +25,9 @@ git = "https://github.com/dirvine/maidsafe_types"
 [dependencies.routing]
 git = "https://github.com/dirvine/routing"
+
+[dependencies.self_encryption]
```
I do not think we need self_encryption as a dependency here.

krishnaIndia replied:
It was not compiling at my end. Will check one more time, removing the dependency

dirvine commented on src/lib.rs:
```diff
@@ -10,6 +10,10 @@ extern crate sodiumoxide;
 extern crate rustc_serialize;
 extern crate cbor;
 extern crate time;
+extern crate bchannel;
```
Not sure bchannel is required either.

krishnaIndia replied:
I might have messed it while merging. I ll check this along with the self_encryption dependency.

krishnaIndia commented:
Tests have been integrated. But there are three failing tests.

krishnaIndia commented:
The failing tests are fixed. All tests are passing for ChunkStore

dirvine commented:
Nice @krishnaIndia  Thanks
